src/components/vr_cooperation/src/vr_module.cc

```diff
@@ -306,23 +232,44 @@ void VRModule::SendMessageToHMI(application_manager::MessagePtr msg) {
   service()->SendMessageToHMI(msg);
   request_controller_.DeleteRequest(msg->correlation_id());
 }
-void VRModule::ReceiveMessageFromHMI() {
+
+void VRModule::SendMessageToHMI(const vr_hmi_api::ServiceMessage& message) {
   LOG4CXX_AUTO_TRACE(logger_);
-  // TODO(Giang):
+  std::string str;
+  google::protobuf::TextFormat::PrintToString(message, &str);
+  LOG4CXX_DEBUG(logger_, "Protobuf message to HMI: " << str);
+  proxy_.Send(message);
 }
+
 void VRModule::SendMessageToMobile(application_manager::MessagePtr msg) {
   LOG4CXX_DEBUG(logger_, "Message to mobile: " << msg->json_message());
   service()->SendMessageToMobile(msg);
   request_controller_.DeleteRequest(msg->correlation_id());
 }
-void VRModule::ReceiveMessageFromMobile() {
-  LOG4CXX_AUTO_TRACE(logger_);
-  // TODO(Giang):
-}
 
 void VRModule::OnReceived(const vr_hmi_api::ServiceMessage& message) {
   LOG4CXX_AUTO_TRACE(logger_);
-  // TODO(KKarlash): Should be implemented
+  std::string str;
+  google::protobuf::TextFormat::PrintToString(message, &str);
+  LOG4CXX_DEBUG(logger_, "Protobuf message: " << str);
+
+  switch (message.rpc_type()) {
+    case vr_hmi_api::MessageType::NOTIFICATION: {
+      if (vr_hmi_api::RPCName::ON_DEACTIVATED == message.rpc()) {
+        commands::OnServiceDeactivatedNotification notification(this);
+        // TODO(Giang): Un-comment when OnDefaultServiceChosen implement
```

Comment:
this code don't relate to OnDefaultServiceChosen implementation
we will implement OnDefaultServiceChosen in else block